src/extension/options-page/DashboardDialogs/Contact.tsx

```diff
@@ -60,22 +65,34 @@ export function DashboardContactDeleteConfirmDialog(
     )
 }
 
-export function DashboardContactDialog(props: WrappedDialogProps<ContactProps & { onUpdated: () => void }>) {
+export function DashboardContactDialog(
+    props: WrappedDialogProps<
+        ContactProps & {
+            onUpdated: (mutatedContact: MutatedContact) => void
+        }
+    >,
+) {
     const { t } = useI18N()
     const classes = useStyles()
     const { contact, onUpdated } = props.ComponentProps!
     const [nickname, setNickname] = useState(contact.nickname)
     const [avatarURL, setAvatarURL] = useState(contact.avatar)
-
-    // TODO: revalidate SWR after this change
-    const onSubmit = useSnackbarCallback(
-        () => Services.Identity.updateProfileInfo(contact.identifier, { nickname, avatarURL, forceUpdateAvatar: true }),
+    const updateProfileInfo = useSnackbarCallback(
+        () =>
+            Services.Identity.updateProfileInfo(contact.identifier, {
+                nickname,
+                avatarURL,
+                forceUpdateAvatar: true,
+            }),
         [nickname, avatarURL],
-        () => {
-            props.onClose()
-            onUpdated()
-        },
     )
+
+    const onSubmit = () => {
+        updateProfileInfo()
+        props.onClose()
+        setTimeout(() => onUpdated({ identifier: contact.identifier, nickname: nickname, avatarURL }), 0)
```

Comment:
There's perhaps a DOM render event loop issue when calling `mutate(mutationData, false)` which casues a UI problem:

![image](https://user-images.githubusercontent.com/63733714/95427594-629c4280-097a-11eb-81c5-30d153112e55.png)
![image](https://user-images.githubusercontent.com/63733714/95427661-8069a780-097a-11eb-9a8b-dfe26546ed3b.png)

Using `setTimeout` to solve it.